Create sms-counter.user.js

plocket opened the pull request:
Allow user to see the number of sms messages the text they've entered will send.

Original is here: https://github.com/plocket/sms-counter

plocket committed:
Create sms-counter.user.js

Allow user to see the number of sms messages the text they've entered will send.

plocket committed:
Add SMS counter bookmarklet

michaelaltmann commented on sms-counter.user.js:
```diff
@@ -0,0 +1,95 @@
+  if (gyrContainer === null) {
+    // Could be on the "Take Action" page
+    gyrContainer = document.querySelector( '#hub_take_action_form_message_body' ).parentElement;
+    console.log(`The text area on the Take Action screen is: ${ gyrContainer }`)
```
Maybe remove this debugging log

michaelaltmann reviewed (commented):
There is something amiss with your math. At 159 characters it says 1 message and 1 char to go.  Then at 160 chars it says 1 message, 160 chars to go.  At 161 characters, it says 2 message and 133 to go.
I think the statement for 160 chars should say 1 message, 134 till the next.  
<img width="545" alt="Screen Shot 2021-03-11 at 8 47 41 AM" src="https://user-images.githubusercontent.com/2830939/110805563-d2269e80-8246-11eb-9d99-497b6468c568.png">
<img width="581" alt="Screen Shot 2021-03-11 at 8 47 52 AM" src="https://user-images.githubusercontent.com/2830939/110805567-d2bf3500-8246-11eb-85f2-b663f60df904.png">
<img width="615" alt="Screen Shot 2021-03-11 at 8 48 07 AM" src="https://user-images.githubusercontent.com/2830939/110805568-d357cb80-8246-11eb-863c-8d42bee1ea6d.png">

plocket commented:
> There is something amiss with your math.

1. Nice eye
2. That actually brings up a discussion topic. Other than some specific complications, 158 is the number that splits messages, but I think some services try to keep whole words as one piece. Should we be cautious and create a margin, or should we be exact?

plocket committed:
Remove debugging console logs

plocket commented:
I'm not sure the statement for 160 quite makes sense that way either now that I think about it. At 160 characters you have one full message. One more character will turn it into two messages. Other options:

#### Option 1
1 msgs (2 chars left)
Num chars 159

1 msgs (1 chars left)
Num chars 160

2 msgs (134 chars left)
Num chars 161

---

#### Option 2
1 msgs (1 chars left)
Num chars 159

1 msgs (0 chars left)
Num chars 160

2 msgs (133 chars left)
Num chars 161

---

Option 2 makes the most sense to me [at the moment].

plocket committed:
Implement new behavior for message count display

This is implementing option 2 of the ones I listed. Expected behavior:
* 159 chars written -> 1 char remaining
* 160 chars written -> 0 chars remaining
* 161 chars written -> 133 chars remaining

Also renames some vars that my brain wasn't catching before.

plocket commented:
I gave it a shot. Let me know what you think.
